volttron/platform/auth.py

```diff
@@ -131,6 +147,196 @@ def setup_zap(self, sender, **kwargs):
         if self.core.messagebus == 'rmq':
             self.vip.peerlist.onadd.connect(self._check_topic_rules)
 
+    def get_entry_rpc_method_authorizations(self, peer_id):
+        rpc_method_authorizations = {}
+        peer_methods = []
+        try:
+            peer_methods = self.vip.rpc.call(
+                peer_id, 'inspect').get(timeout=4)["methods"]
+        except gevent.Timeout:
+            _log.warn(f'{peer_id} has timed out while attempting to get rpc methods')
+        except Unreachable:
+            _log.warn(f'{peer_id} is unreachable while attempting to get rpc methods')
+        except MethodNotFound as e:
+            _log.error(e)
+
+        for method in peer_methods:
+            try:
+                rpc_method_authorizations[method] = \
+                    self.vip.rpc.call(peer_id, 'get_rpc_authorizations', method).get(timeout=4)
```

Comment:
Why do we need to make rpc call to agent for every method in the inspect list? This is a costly operation since it needs to be done for every agent in the auth list.

Reply:
Fixed